libs/community/langchain_community/tools/azure_cognitive_services/content_safety.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import annotations
+
+import logging
+import os
+from typing import Any, Dict, Optional
+
+from langchain_core.callbacks import CallbackManagerForToolRun
+from langchain_core.tools import BaseTool
+
+logger = logging.getLogger(__name__)
+
+
+class AzureContentSafetyTextTool(BaseTool):
+    """
+    A tool that interacts with the Azure AI Content Safety API.
+
+    This tool queries the Azure AI Content Safety API to analyze text for harmful
+    content and identify sentiment. It requires an API key and endpoint,
+    which can be set up as described in the following guide:
+
+    https://learn.microsoft.com/python/api/overview/azure/ai-contentsafety-readme?view=azure-python
+
+    Attributes:
+        content_safety_key (str):
+            The API key used to authenticate requests with Azure Content Safety API.
+        content_safety_endpoint (str):
+            The endpoint URL for the Azure Content Safety API.
+        content_safety_client (Any):
+            An instance of the Azure Content Safety Client used for making API
+            requests.
+
+    Methods:
+        _sentiment_analysis(text: str) -> Dict:
+            Analyzes the provided text to assess its sentiment and safety,
+            returning the analysis results.
+
+        _run(query: str,
+            run_manager: Optional[CallbackManagerForToolRun] = None) -> str:
+            Uses the tool to analyze the given query and returns the result.
+            Raises a RuntimeError if an exception occurs.
```

Comment:
Given these methods are prefixed with an underscore / marked internal, we should not add them to the public documentation.